app/Models/Actionlog.php

```diff
@@ -450,6 +444,62 @@ public function determineActionSource(): string
 
     }
 
+    public function uploads_file_url()
```

Comment:
These `switch()` statements are fine, but it feels like we have some array maps up in the AssetFilesController that are similar or related to these - should those maps go here in the Actionlog as well, just to keep related things in the same general related place? I could also see the argument for those _other_ maps being close to the controller where they’re consumed, so I’m not 100% on this.

You could maybe make those maps be static variables on the Actionlog if you like?

Reply:
I'd actually prefer them to be static under the models themselves and/or make a trait for `CanHaveUploads` or some such, but this PR seemed a step in the right direction.